config/element-api.php

```diff
@@ -166,18 +166,84 @@ function getFundingProgrammes($locale)
             'status' => 'live',
         ],
         'transformer' => function (Entry $entry) use ($locale) {
+            
+            
+
             return [
                 'id' => $entry->id,
                 'status' => $entry->status,
                 'title' => $entry->title,
                 'url' => $entry->url,
                 'urlPath' => $entry->uri,
-                'content' => getFundingProgramMatrix($entry, $locale),
+                'content' => getFundingProgramMatrix($entry, $locale)
             ];
         },
     ];
 }
 
+
+function getDraftOrVersionOfEntry($entry) {
+    $status = 'live';
+
+    $isDraft = \Craft::$app->request->getParam('draft');
+    $isVersion = \Craft::$app->request->getParam('version');
+
+    if (!$isDraft && !$isVersion) {
+        return false;
+    }
+    
+    if ($isDraft) {
+        $status = 'draft';
+        $revisionId = $isDraft;
+        $revisionMethod = 'getDraftsByEntryId';
+        $entryRevisionMethod = 'getDraftById';
+        $revisionIdParam = 'draftId';
+    } else if ($isVersion) {
+        $status = 'version';
+        $revisionId = $isVersion;
+        $revisionMethod = 'getVersionsByEntryId';
+        $entryRevisionMethod = 'getVersionById';
+        $revisionIdParam = 'versionId';
+    }
+
+    if ($revisionId) {
+
+        // Check they're allowed to access this content
+        $user = Craft::$app->user->getIdentity();
+        if (!$user || (!$user->admin && !$user->isInGroup('authors'))) {
+            // @TODO
```

Comment:
Thoughts welcome here on what sort of auth we want to do here.

Reply:
At the API throwing the exception you have commented out seems fine. On the app end we might want to do something more graceful.

Reply:
It's more about what criteria that exception should be thrown at (eg. `(!$user->admin && !$user->isInGroup('authors'))`) – not sure if we need a group or admin status etc.